src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/MethodLdtokenNode.cs

```diff
@@ -0,0 +1,181 @@
+        {
+            return Array.Empty<CombinedDependencyListEntry>();
+        }
+
+        public override bool HasDynamicDependencies
+        {
+            get
+            {
+                // Only generic virtual methods work have dynamic dependencies
+                if (_method.HasInstantiation && _method.IsVirtual)
+                {
+                    if (_method.IsCanonicalMethod(CanonicalFormKind.Specific))
+                        return false;
+
+                    if (_method.OwningType.ContainsGenericVariables || _method.ContainsGenericVariables)
+                        return false;
+
+                    if (_method.OwningType.IsCanonicalSubtype(CanonicalFormKind.Universal) &&
+                        _method.OwningType != _method.OwningType.ConvertToCanonForm(CanonicalFormKind.Universal))
+                        return false;
+
+                    return true;
+                }
+                return false;
+            }
+        }
+
+        public override IEnumerable<CombinedDependencyListEntry> SearchDynamicDependencies(List<DependencyNodeCore<NodeFactory>> markedNodes, int firstNode, NodeFactory factory)
+        {
+            Debug.Assert(_method.IsVirtual && _method.HasInstantiation);
+
+            List<CombinedDependencyListEntry> dynamicDependencies = new List<CombinedDependencyListEntry>();
+
+            for (int i = firstNode; i < markedNodes.Count; i++)
+            {
+                DependencyNodeCore<NodeFactory> entry = markedNodes[i];
+                EETypeNode entryAsEETypeNode = entry as EETypeNode;
+
+                if (entryAsEETypeNode == null)
+                    continue;
+
+                TypeDesc potentialOverrideType = entryAsEETypeNode.Type;
+                if (!(potentialOverrideType is DefType))
+                    continue;
+
+                Debug.Assert(!potentialOverrideType.IsRuntimeDeterminedSubtype);
+
+                if (_method.OwningType.HasSameTypeDefinition(potentialOverrideType) && potentialOverrideType.IsInterface && (potentialOverrideType != _method.OwningType))
+                {
+                    if (_method.OwningType.CanCastTo(potentialOverrideType))
+                    {
+                        // Variance expansion
+                        MethodDesc matchingMethodOnRelatedVariantMethod = potentialOverrideType.GetMethod(_method.Name, _method.GetTypicalMethodDefinition().Signature);
+                        matchingMethodOnRelatedVariantMethod = _method.Context.GetInstantiatedMethod(matchingMethodOnRelatedVariantMethod, _method.Instantiation);
+                        dynamicDependencies.Add(new CombinedDependencyListEntry(factory.MethodLdtoken(matchingMethodOnRelatedVariantMethod), null, "GVM Variant Interface dependency"));
+                    }
+                }
+
+                // Open generic types are not interesting.
+                if (potentialOverrideType.ContainsGenericVariables)
```

Comment:
There is no need to check for this... There should never be any open generics at this level